client/my-sites/plans/jetpack-plans/product-store/utils/get-sorted-displayable-products.ts

```diff
@@ -0,0 +1,35 @@
+import {
+	JETPACK_ANTI_SPAM_PRODUCTS,
+	JETPACK_BACKUP_PRODUCTS,
+	JETPACK_BOOST_PRODUCTS,
+	JETPACK_CRM_PRODUCTS,
+	JETPACK_SCAN_PRODUCTS,
+	JETPACK_SEARCH_PRODUCTS,
+	JETPACK_VIDEOPRESS_PRODUCTS,
+} from '@automattic/calypso-products';
+import { SelectorProduct } from '../../types';
+
+const setProductsInPosition = ( slugs: ReadonlyArray< string >, position: number ) =>
+	slugs.reduce( ( map, slug ) => ( { ...map, [ slug ]: position } ), {} );
+
+const DISPLAYABLE_PRODUCT_POSITION_MAP: Record< string, number > = {
+	...setProductsInPosition( JETPACK_BACKUP_PRODUCTS, 0 ),
+	...setProductsInPosition( JETPACK_ANTI_SPAM_PRODUCTS, 1 ),
+	...setProductsInPosition( JETPACK_SCAN_PRODUCTS, 2 ),
+	...setProductsInPosition( JETPACK_VIDEOPRESS_PRODUCTS, 3 ),
+	...setProductsInPosition( JETPACK_SEARCH_PRODUCTS, 4 ),
+	...setProductsInPosition( JETPACK_BOOST_PRODUCTS, 5 ),
+	// TODO - Set Jetpack social product order here
+	...setProductsInPosition( JETPACK_CRM_PRODUCTS, 7 ),
+};
+
+const sortByProductPosition = ( productA: SelectorProduct, productB: SelectorProduct ) => {
+	return (
+		DISPLAYABLE_PRODUCT_POSITION_MAP[ productA.productSlug ] -
+		DISPLAYABLE_PRODUCT_POSITION_MAP[ productB.productSlug ]
+	);
+};
+
+export const getSortedDisplayableProducts = ( products: SelectorProduct[] ): SelectorProduct[] => {
+	return [ ...products ].sort( sortByProductPosition );
+};
```

Comment:
What about creating an array or Set of products as `DISPLAYABLE_PRODUCTS` with the order set to the way they should be displayed. Then you can filter out the ones which are not passed in `products` and then map through them using `slugToSelectorProduct`.

Reply:
Hi @manzoorwanijk I was hoping to avoid using array or set and use a map to get the position in constant time. So I can achieve `o(n^2)` time complexity with the generic sort algorithm.